charts/Makefile

```diff
@@ -37,6 +32,15 @@ WKDIR ?= ../out
 # Define the build target for each project
 define make-chart-target
 build-$(1)-chart:
+	$(eval BRANCH := $(if $(BRANCH_$(1)),$(BRANCH_$(1)),$(BRANCH)))
```

Comment:
I think we've outgrown the Makefile approach.
This is starting to rely on Makefile quirks, which makes it very difficult to read and maintain. We have introduced two components that do not comply with the assumptions taken so far.

I guess this normally will be run with BRANCH=release-2.9 or BRANCH=master, however now it cannot be the case any longer. Overrides are supported only in-makefile, to override specific components from ENV we should improve the current solution.

For now this should work but I believe we should consider changing approach.

Reply:
In fact, I was doubting make those changes for CAPOA first PR. The work in this [PR](https://github.com/stolostron/cluster-api-installer/pull/47/files) to avoid repeating the charts make tags end up more code to write and becomes hard to follow. Can we have make tag for every chart (as it was before) and simply pass the env varaible based on the chart tag.

Reply:
Reverted to a similar format of the original.
It's still an issue, as this approach it's extremely error prone, however it's now more understandable.

@serngawy how is this makefile being run? Do we expect to be able to override from env vars? Or is it always executed as "make build-helm-charts"?

Reply:
Usually we run it as "make build-helm-charts" for ex; the sync job [here](https://github.com/stolostron/cluster-api-installer/blob/main/.github/workflows/sync-providers.yaml#L38) run weekly to catch any changes happen in origin repo.